### PR TITLE
[Innawood] More items repairable

### LINKED ISSUE #54270: [Innawood] Make repairing more things possible
### Is your feature request related to a problem? Please describe.

Currently, in the Innawood mod, the only repairing tools that can be crafted and used are the wood, bone, and curved needles. This means that only cloth (i.e. wool, cotton, leather, fur) items can be repaired. Thanks to #54149, cotton items can be repaired. However, this still leaves items made out of metal, wood, paper, plastic, or chitin unrepairable. This makes chitin or metal armor and metal weapons unviable to use. 

### Solution you would like.

Repairing wood, paper, and chitin would be possible if the misc repair kit is obtainable. Currently, it requires an x-acto knife, which is loot only. This could be fixed by making the x-acto knife craftable, removing it from the requirements, or requiring some other craftable item such as a survival knife or makeshift knife. 

As for metal items, I'm not sure what the solution should be, since they would normally require an arc welder, which doesn't seem feasible to craft from scratch in an innawoods scenario. Perhaps a new item like a "blacksmith's repair kit" could be added, similar to a firearm repair kit in that it is a combination of many other tools. 

The only craftable plastic item I could find that would need repairing is the safety glasses. The recipe for these has been changed to require glass instead of plastic, so they should be made out of the glass material. 

### Describe alternatives you have considered.

If plastic chunks are made craftable by making bleach or vinegar obtainable, it might be possible to craft more useful plastic items, in which case some other solution would be needed to repair those items. 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by juliavetter**
#### Summary
Mods "[Innawood] More items repairable"

#### Purpose of change

Fixes #54270.

#### Describe the solution

The x-acto knife is now craftable, with the same recipe as the carving knife. This allows one to craft a misc repair kit, which can repair more types of items. It also changes the safety glasses material to glass, since the innawood crafting recipe uses glass. The blacksmith repair kit is added to repair metal items. It consumes charcoal and is roughly comparable to a makeshift arc welder. It can be crafted with a metal fileset, flatjaw tongs, hammer, and a clay canister. 

#### Describe alternatives you've considered

- Change the misc repair kit recipe to use a carving knife or some other tool
I think this fix is the most simple way to add this functionality to innawood. 
- Make safety glasses repairable
It doesn't seem feasible to repair a pair of glasses with primitive materials. Even with modern technology, it's probably easier to just make a new piece of glass. 

#### Testing

- Confirmed that the x-acto knife is craftable, and that it has the same recipe as the paring knife. 
- Confirmed that safety glasses are made out of glass and not repairable. 
- Confirmed that the blacksmith repair kit is craftable and does work similarly to an arc welder to repair metal items. 

#### Additional context

@Light-Wave Let me know what you think of these changes.

This is also my first pull request, so please let me know if I've done anything wrong!

### CODE REVIEW

**Comment by VoleurVolant** [comment]
I would strongly suggest removing or altering the crafting recipe instead. The entire point of safety glasses is to function as eye protection in a workshop setting, and as such, they are made of shatter-resistant polycarbonate. Internal consistency is important, but this change is a move away from both realism and verisimilitude.

**Comment by kurtisvetter** [comment]
> I would strongly suggest removing or altering the crafting recipe instead. The entire point of safety glasses is to function as eye protection in a workshop setting, and as such, they are made of shatter-resistant polycarbonate. Internal consistency is important, but this change is a move away from both realism and verisimilitude.

I see what you're saying. I think the main reason it was changed in the first place was to allow the basic chemistry set to be craftable in innawood. In that case, it might make more sense to remove their requirement from the basic chemistry set. Or, we could leave it as is and also reduce their protection values. If they are made of glass, they should have the same protection values as other glasses in the game. 

The only other thing you can craft with them in innawood is swim goggles. There's no reason as far as I know why swim goggles need to be plastic rather than glass, so it's fine for them to be glass in that situation.

**Comment by Proxima-b** [comment]
I think innawood mod should just remove all book requirements for all recipes, so that players can craft just about anything. Afterall the whole point of this mod is to make everything with nothing

**Comment by Proxima-b** [comment]
> I would strongly suggest removing or altering the crafting recipe instead. The entire point of safety glasses is to function as eye protection in a workshop setting, and as such, they are made of shatter-resistant polycarbonate. Internal consistency is important, but this change is a move away from both realism and verisimilitude.

That is not true, There actually are pairs of glass safety glasses at my workplace so that's it.

File: data/mods/innawood/items/armor_eyewear.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+[
+  {
+    "id": "glasses_safety",
+    "type": "ARMOR",
+    "name": { "str": "pair of safety glasses", "str_pl": "pairs of safety glasses" },
+    "description": "A pair of plastic glasses, used in workshops, sports, chemistry labs, and many other places.  Provides great protection from damage.",
```
Comment (Light-Wave) [inline data/mods/innawood/items/armor_eyewear.json:6]: Suggest removing the word plastic

**Comment by Light-Wave** [comment]
> I would strongly suggest removing or altering the crafting recipe instead. The entire point of safety glasses is to function as eye protection in a workshop setting, and as such, they are made of shatter-resistant polycarbonate. Internal consistency is important, but this change is a move away from both realism and verisimilitude.

Seems to me glass should be useful enough when dealing with chemicals. At least about as good as cutting up a plastic bottle and using that instead. I suppose an alternative would be to have a "glass safety goggles" item that works for making chemistry sets, but isn't good enough to help you in a workshop. But that seems a bit excessive. Especially for a mod.

File: data/mods/innawood/items/armor_eyewear.json (hunk 1 above)
Comment (kurtisvetter) [reply data/mods/innawood/items/armor_eyewear.json]: Thanks!

**Comment by Light-Wave** [comment]
Either way, this pull request looks good to me. It has my approval.

**Review by Light-Wave — APPROVED** [review]

**Comment by NetSysFire** [comment]
This is very meta, but there is so much discussion going on about the Innawood mod, which is not bad at all, how about a dedicated github discussion for that? I created one for you (since currently normal users can not create one easily): https://github.com/CleverRaven/Cataclysm-DDA/discussions/54291

**Review by Light-Wave — APPROVED** [review]

File: data/mods/innawood/items/tool_workshop.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      }
+    ],
+    "flags": [ "ALLOWS_REMOTE_USE" ],
+    "pocket_data": [
+      {
+        "pocket_type": "MAGAZINE",
+        "rigid": true,
+        "ammo_restriction": { "charcoal": 100 }
+      }
+    ]
```
Comment (kevingranade) [inline data/mods/innawood/items/tool_workshop.json:47]:
```suggestion
    "pocket_data": [ { "pocket_type": "MAGAZINE", "rigid": true, "ammo_restriction": { "charcoal": 100 } } ]
```

**Comment by Zireael07** [comment]
Creator of chemistry set here, I never intended for the "safety glasses" to be either polycarbonate or glass specifically, either is fine.

**Comment by kurtisvetter** [comment]
I think this is ready to merge.

**Pull request merged**